src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -230,6 +231,30 @@ private Callback kafkaSendCallback(final BatchItem item, final CountDownLatch do
     }
 
     @Override
+    public Map<String, Long> materializePositions(final String topicId, String position) throws ServiceUnavailableException {
+        try (final Consumer<String, String> consumer = kafkaFactory.getConsumer()) {
+            final org.apache.kafka.common.TopicPartition[] kafkaTPs = consumer
+                    .partitionsFor(topicId)
+                    .stream()
+                    .map(p -> new org.apache.kafka.common.TopicPartition(topicId, p.partition()))
+                    .toArray(org.apache.kafka.common.TopicPartition[]::new);
+            if (position.equals(Cursor.BEFORE_OLDEST_OFFSET)) {
+                consumer.seekToBeginning(kafkaTPs);
+            } else if (position.equals(Cursor.AFTER_NEWEST_OFFSET)) {
+                consumer.seekToEnd(kafkaTPs);
+            } else {
+                throw new ServiceUnavailableException("Bad offset specification " + position + " for topic " + topicId);
```

Comment:
Agree, on this level of processing I can't throw it. I'll wait for subscription creation endpoint and will 
use enumeration that is there and this will fix all problems.